client/src/app/AppParent.js

```diff
@@ -292,12 +290,33 @@ export default class AppParent extends PureComponent {
 
   /**
    *
-   * @param {string} errorMessage
    * @param {Error} error
-   * @param {Tab} tab
+   * @param {Tab} [tab]
    */
-  logToBackend(errorMessage, error, tab) {
-    this.props.globals.log.error(errorMessage, error, tab);
+  logToBackend(error, tab) {
+    const entry = this.getBackendLogEntry(error, tab);
+
+    this.props.globals.log.error(entry);
+  }
+
+  /**
+   * Creates entry transferrable to backend.
+   * @param {Error} error
+   * @param {Tab} [tab]
+   *
+   * @returns {{ message: string, stack: string }}
+   */
+  getBackendLogEntry(error, tab) {
+    const entry = {
+      message: this.getEntryMessage(error.message, tab),
+      stack: error.stack
+    };
+
+    return entry;
+  }
+
+  getEntryMessage(message, tab) {
+    return tab ? `[${tab.file.name}] ${message}` : message;
```

Comment:
Let's log the full file path, if available here, too.

And handle the case that a tab does not have a referenced file (empty tab or some extension tab)